src/unlimited.cpp

```diff
@@ -1053,21 +1054,23 @@ UniValue settrafficshaping(const UniValue& params, bool fHelp)
 // This way we avoid having to lock cs_main so often which tends to be a bottleneck.
 void IsChainNearlySyncdInit() 
 {
-    LOCK2(cs_main, cs_ischainnearlysyncd);
-    if (!pindexBestHeader) fIsChainNearlySyncd = false;  // Not nearly synced if we don't have any blocks!
-    else
-      {
-      if(chainActive.Height() < pindexBestHeader->nHeight - 2)
-        fIsChainNearlySyncd = false;
-      else
-        fIsChainNearlySyncd = true;
-      }
+    bool flag = fIsChainNearlySyncd.load();
+    LOCK(cs_main);
+    if (!pindexBestHeader) {
+        // Not nearly synced if we don't have any blocks!
+        while (flag && !fIsChainNearlySyncd.compare_exchange_weak(flag, false));
```

Comment:
I'm not sure I understand the whole flow here. Basically, we want fIsChainNearlySyncd to be true if are within 2 block of the best header we know of.

I'm not sure when this function is called so it's difficult to assert correctness or not.